### PR TITLE
Fixed prefix and destination directory for CPAN

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/flatpak/flatpak-builder-tools/issues/273

### CODE REVIEW

File: cpan/flatpak-cpan-generator.pl [hunk 1]
```diff
@@ -122,7 +122,7 @@ sub main {
       "    mod_dir=\$1",
       "    cd \$mod_dir",
       "    if [ -f 'Makefile.PL' ]; then",
-      "        perl Makefile.PL && make install",
+      "        perl Makefile.PL PREFIX=/ && make install DESTDIR=/app",
```
Comment (TingPing) [inline cpan/flatpak-cpan-generator.pl:125]: This sounds wrong. PREFIX would be `/app`, DESTDIR should be unset.

Comment (TingPing) [reply cpan/flatpak-cpan-generator.pl:125]: Technically PREFIX is `/usr` if its a runtime and `/app` for applications, this script can, although probably isn't, usable for both.

Comment (Mailaender) [reply cpan/flatpak-cpan-generator.pl:125]: It is counter intuitive, but this is what I get:

> Installing /app/lib/perl5/site_perl/5.34.0/x86_64-linux/auto/Compress/Raw/Lzma/Lzma.so
Installing /app/lib/perl5/site_perl/5.34.0/x86_64-linux/auto/Compress/Raw/Lzma/autosplit.ix
Installing /app/lib/perl5/site_perl/5.34.0/x86_64-linux/Compress/Raw/Lzma.pm
Installing /app/share/man/man3/Compress::Raw::Lzma.3
Appending installation info to /app//lib/perl5/5.34.0/x86_64-linux/perllocal.pod

Working test case https://github.com/Mailaender/flathub/tree/010editor

Comment (TingPing) [reply cpan/flatpak-cpan-generator.pl:125]: I'm not a perl person but here is how these variables work in everything unix I've ever used.

- PREFIX is the actual prefix of the runtime environment, it is often passed to actual source code and is very relevant.
- DESTDIR is a prefix to copy files into, it is used only at install time and the purpose of it is for intermediate installation steps before final packaging. This does not apply to flatpak.

So while its possible Perl is doing something very odd. This just looks incorrect and risks the wrong PREFIX being used.

Comment (Mailaender) [reply cpan/flatpak-cpan-generator.pl:125]: What do you suggest instead to write into the correct directories?

Comment (hfiguiere) [reply cpan/flatpak-cpan-generator.pl:125]: `PREFIX=${FLATPAK_DEST}`

keep `DESTDIR` unset

Comment (TingPing) [reply cpan/flatpak-cpan-generator.pl:125]: Are you setting prefix during both build and install?

Comment (Mailaender) [reply cpan/flatpak-cpan-generator.pl]: That's it! Confirmed it works.

**Pull request merged**